lib/maze/timers.rb

```diff
@@ -0,0 +1,61 @@
+module Maze
+
+  # A simple run/stop timer
+  class Timer
+    attr_accessor :total
+
+    def initialize
+      @started = 0
+      @total = 0
+      @running = false
+    end
+
+    def run
+      raise 'Timer already running' if @running
```

Comment:
I put these in to make it obvious if a client tries to use it in the wrong way, but it could be asking for trouble if there are error conditions that, for example, prevent a timer being stopped after it is run.  I might be better logging and error and flagging at the end that the timer should not be trusted?

Reply:
Would a block-based API work? That way you can avoid the timer getting into an invalid state e.g.

```rb
class Timer
  attr_accessor :total

  def initialize
    @total = 0
  end

  def time(&block)
    start = Time.now

    block.call

    @total += Time.now - start
  end
end
```

It might also be useful to return the result of the block so you can measure code that needs to return something

Reply:
As I was putting the calls in I knew there must be a better way of doing it.  Would the last need need to go in an `ensure` block?

Reply:
Yeah, it probably should do. That would also make it return the result of the block as well

Reply:
Done, that makes it much cleaner.